InventoryService/cmd/controllers/item_controller.go

```diff
@@ -0,0 +1,263 @@
+		passed = false
+	}
+
+	if item.Alcohol < 0 || item.Alcohol > 1 {
+		errorMessage += resources.InadequateAlcohol + " "
+		passed = false
+	}
+
+	if !(item.MeasurmentUnit == "pcs" || item.MeasurmentUnit == "g" || item.MeasurmentUnit == "ml" || item.MeasurmentUnit == "") {
+		errorMessage += resources.WrongMeasurement + " "
+		passed = false
+	}
+	return
+}
+
+// @BasePath /api/v1
+
+// GetIngridientsList godoc
+// @Summary        Gets list of all ingridients
+// @Description    Get complete list of all ingridients availible for user
+// @Tags           Inventory manipulation
+// @Produce        application/json
+// @Success        200
+// @Router         /inventory [get]
+func (controller *ItemController) GetIngridientsList(c *gin.Context) {
+	ingridients, err := controller.Client.Ingridient.
+		Query().
+		All(c)
+
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusInternalServerError, err.Error(), nil)
+		return
+	}
+
+	resources.ResponseJSON(c, http.StatusOK, "", ingridients)
+}
+
+// @BasePath /api/v1
+
+// GetIngridientsList godoc
+// @Summary         Gets list ingridients by filter
+// @Description     Get list of ingridients filtering by it's name and/or parameters
+// @Tags            Inventory manipulation
+// @Produce         application/json
+// @Param           item query models.ItemModelFilters true "Item to search for"
+// @Success         200
+// @Failure         400
+// @Router          /ingridient/search [get]
+func (controller *ItemController) GetIngridientByFilter(c *gin.Context) {
+	var filters models.ItemModelFilters
+	filters.AlcoholMax = 1
+
+	err := c.BindQuery(&filters)
+
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusBadRequest, err.Error(), nil)
+		return
+	}
+
+	if filters.AlcoholMin < 0 || filters.AlcoholMax > 1 {
+		resources.ResponseJSON(c, http.StatusBadRequest, resources.InadequateAlcohol, nil)
+		return
+	}
+
+	ingridients, err := controller.Client.Ingridient.
+		Query().
+		Where(
+			ingridient.And(
+				ingridient.NameContainsFold(filters.Name),
+				ingridient.AlcoholGT(filters.AlcoholMin),
+				ingridient.AlcoholLT(filters.AlcoholMax),
+				ingridient.IsDry(filters.IsDry),
+			),
+		).
+		All(c)
+
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusInternalServerError, err.Error(), nil)
+		return
+	}
+
+	resources.ResponseJSON(c, http.StatusOK, "", ingridients)
+}
+
+// @BasePath /api/v1
+
+// AddIngridient godoc
+// @Summary           Adds new ingridient
+// @Description       Add new ingridient to database
+// @Tags              Inventory manipulation
+// @Accept            json
+// @Produce           application/json
+// @Param             item body models.ItemModelCreate true "Item data"
+// @Success           201
+// @Failure           400
+// @Router            /ingridient [post]
+func (controller *ItemController) AddIngridient(c *gin.Context) {
+	jsonData, err := c.GetRawData()
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusBadRequest, err.Error(), nil)
```

Comment:
Empty error message?

Reply:
No, empty error data object. Message is err.Error()

Reply:
Understood, sorry